warp/Network/Wai/Handler/Warp/Header.hs

```diff
@@ -19,27 +19,35 @@ indexRequestHeader hdr = traverseHeader hdr requestMaxIndex requestKeyIndex
 
 idxContentLength,idxTransferEncoding,idxExpect :: Int
 idxConnection,idxRange,idxHost :: Int
-idxContentLength    = 0
```

Comment:
Not specifically relevant to this change, but I wonder if we can get the same performance with easier to maintain code if we create an enum type here instead of using the `Int`s directly.